### PR TITLE
ENH: Add Fio stone

### PR DESCRIPTION
**Pull request by julio-santiesteban**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -105,10 +106,23 @@ func main() {
 			Scheme:        mgr.GetScheme(),
 			EventRecorder: newEventRecorder(clientSet),
 		},
-		Log: ctrl.Log.WithName("controllers").WithName("iperf3")}).SetupWithManager(mgr); err != nil {
+		Log: ctrl.Log.WithName("controllers").WithName("iperf3"),
```
Comment (wgershwitz) [inline main.go]: s/iperf3/fio/

Comment (wgershwitz) [reply main.go]: nevermind, looks good :)

File: main.go [hunk 2]
```diff
@@ -105,10 +106,23 @@ func main() {
...
+	}).SetupWithManager(mgr); err != nil {
 		setupLog.Error(err, "unable to create controller", "controller", "Iperf3")
 		os.Exit(1)
 	}
+	if err = (&controllers.FioReconciler{
+		K8S: k8s.Access{
```
Comment (wgershwitz) [inline main.go]: this struct is shared between iperf and fio. could we please use a shared var in both cases?

File: controllers/fio_controller.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	"k8s.io/client-go/tools/reference"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	perfv1alpha1 "github.com/xridge/kubestone/api/v1alpha1"
+	"github.com/xridge/kubestone/pkg/k8s"
+)
+
+// FioReconciler provides fields from manager to reconciler
+type FioReconciler struct {
```
Comment (wgershwitz) [inline controllers/fio_controller.go]: In a follow-up pull request we should create a base class which is shared between every controller.
The current content of FioReconciler is a good target (as it maches iperf reconciler atm).

Comment (julio-santiesteban) [reply controllers/fio_controller.go]: Noted

**Pull request merged**